app/forms/ContractorDetailsForm.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.validation.StringConstraints.{validateChar, validateSize}
+import models.ContractorDetailsViewModel
+import play.api.data.Form
+import play.api.data.Forms.mapping
+import forms.validation.mappings.MappingUtil.trimmedText
+import forms.validation.utils.ConstraintUtil.ConstraintUtil
+import play.api.data.validation.Constraint
+import play.api.data.validation.Constraints.nonEmpty
+
+
+object ContractorDetailsForm {
+
+  def contractorDetailsForm(
+                            isAgent: Boolean
+                           ): Form[ContractorDetailsViewModel] = {
+
+    val contractorName = "contractorName"
+    val employerReferenceNumber = "employerReferenceNumber"
+    val nameCharLimit = 105
+    val nameRegex = "^[0-9a-zA-Z\\\\{À-˿’}\\-&\\/:;., ]{1,105}$"
+    val refRegex = "^\\d{3}\\/[A-Za-z0-9 ]{1,10}$"
```

Comment:
this validation for the employerRef is different to the one on confluence: ^[0-9]{3}\\/[^ ].{0,9}$